Revise troubleshooting guidance

benthorner opened the pull request:
https://trello.com/c/i3Y2vAua/188-documentation-on-how-to-troubleshoot-apps-failing-to-start-in-govuk-docker

This PR is a review of the general troubleshooting guidance
for apps, which we know is unhelpful for some people.

Please see the commits for more details.

ChrisBAshton commented on docs/troubleshooting.md:
````diff
@@ -71,17 +69,7 @@ scutil --dns
-
-```
-bin/doctor
-```
````
May be worth linking to https://github.com/alphagov/govuk-docker/issues/435 in your commit, as I wouldn't have known why this section was removed if I hadn't already read your issue..

benthorner replied:
Ah, so I was wondering whether to be explicit about this in the commit: the section still exists at the top - it's just de-duplicated. Depending on what happens with #435, we may yet remove or edit it, but I agree that would be premature here.

Is that OK?

ChrisBAshton replied:
Ah, didn't spot that. Makes sense. Thanks

ChrisBAshton commented on docs/troubleshooting.md:
```diff
@@ -12,21 +12,33 @@ bin/doctor
+govuk-docker down
 
-# try clearing all containers / volumes
-govuk-docker rm -sv
```
I know you want to encourage people to diagnose and fix errors 'properly', but sometimes if you're in a bind and need to get something working quickly, this is just the easiest option. I'd really like you to leave it in, even if it's caveated with "You really shouldn't have to do this...!"

benthorner replied:
I agree there is a need to have a clean slate. The `govuk-docker down` command I've suggested is actually equivalent to this, but has the advantage of being easier to explain and write down.

Does that help address your concern?

ChrisBAshton replied:
Sure, thanks for explaining.

ChrisBAshton reviewed (commented):
Have had a read and have a couple of suggestions below.

An aside: one note I made from yours & Huw's talk the other week was `govuk-docker-up app-live` for looking at live data. That may not be the exact command or purpose, but it was new to me and I wrote it down - can't see any further detail about it in the repo though. Is it documented somewhere? :)

benthorner commented:
Thanks for the review @ChrisBAshton. I think I may have addressed your comments with a bit more explanation, but let me know if you'd like me to clarify the commits, for example.

The `app-live` stack has a slightly awkward history, because I think some devs still rely on the `startup.sh` scripts in each repo, which replicate the functionality. Anyway, the functionality in GOV.UK Docker is documented [in the README](https://github.com/alphagov/govuk-docker#the-app--stacks), and should  also now be a little more prominent, since I rewrote it.

ChrisBAshton reviewed (approved):
LGTM 👍
